File: Exemplos SmartPOS/App_IntentDigitalHUB_Java/Elgin_IntentDigitalHUB_SmartPOS_Java-font/app/src/main/java/com/elgin/java_intentdigitalhub_smartpos/IntentDigitalHubService/TERMICA/Commands/ImpressaoTexto.java
package com.elgin.java_intentdigitalhub_smartpos.IntentDigitalHubService.TERMICA.Commands;

import androidx.annotation.NonNull;

import com.elgin.java_intentdigitalhub_smartpos.IntentDigitalHubService.TERMICA.TermicaCommand;
import com.google.gson.JsonObject;

final public class ImpressaoTexto extends TermicaCommand {
    final private String dados;
    final private int posicao;
    final private int stilo;
    final private int tamanho;

    public ImpressaoTexto(@NonNull String dados, int posicao, int stilo, int tamanho) {
        super("ImpressaoTexto");
        this.dados = dados;
        this.posicao = posicao;
        this.stilo = stilo;
        this.tamanho = tamanho;
    }

    @Override
    protected JsonObject functionParametersJson() {
        JsonObject functionParametersJson = new JsonObject();

        functionParametersJson.addProperty("dados", this.dados);
        functionParametersJson.addProperty("posicao", this.posicao);
        functionParametersJson.addProperty("stilo", this.stilo);
        functionParametersJson.addProperty("tamanho", this.tamanho);

        return functionParametersJson;
    }
}

File: Exemplos SmartPOS/App_IntentDigitalHUB_Kotlin/Elgin_IntentDigitalHUB_SmartPOS_Kotlin-font/app/src/main/java/com/elgin/kotlin_intent_digital_hub_smartpos/IntentDigitalHubService/TERMICA/Commands/ImpressaoTexto.kt
package com.elgin.kotlin_intent_digital_hub_smartpos.IntentDigitalHubService.TERMICA.Commands

import com.elgin.kotlin_intent_digital_hub_smartpos.IntentDigitalHubService.TERMICA.TermicaCommand

import com.google.gson.JsonObject


class ImpressaoTexto(
    private val dados: String,
    private val posicao: Int,
    private val stilo: Int,
    private val tamanho: Int)
    : TermicaCommand("ImpressaoTexto") {
    override fun functionParametersJson(): JsonObject {
        val functionParametersJson = JsonObject()
        functionParametersJson.addProperty("dados", dados)
        functionParametersJson.addProperty("posicao", posicao)
        functionParametersJson.addProperty("stilo", stilo)
        functionParametersJson.addProperty("tamanho", tamanho)
        return functionParametersJson
    }
}
